### ISSUE TITLE
Minified JS problem?

### ISSUE THREAD

**Issue by szymonmichalak** [fragaria/angular-daterangepicker#34]
Hi,
first of all - thanks for your great work!

Directive is working perfectly, while testing it locally (using yeoman with angularjs + requirejs, wiredep, loading all dependencies). 

However, during the built, my grunt configuration is performing minification (etc.) and after build such error is printed into console:

```
TypeError: undefined is not a function
    at a.directive.link.n (http://localhost/scripts/main.js:32:1100)
    at a.directive.link (http://localhost/scripts/main.js:32:1311)
    at http://localhost/scripts/main.js:3:5951
    at invokeLinkFn (http://localhost/scripts/main.js:3:6069)
    at nodeLinkFn (http://localhost/scripts/main.js:2:24060)
    at compositeLinkFn (http://localhost/scripts/main.js:2:13776)
    at compositeLinkFn (http://localhost/scripts/main.js:2:13866)
    at $get.node (http://localhost/scripts/main.js:2:12447)
    at compile (http://localhost/scripts/main.js:20:12698)
    at invokeLinkFn (http://localhost/scripts/main.js:3:6069) <div class="date-picker ng-pristine ng-untouched ng-valid ng-isolate-scope" options="datepickerOptions" ng-model="active.date" date-range-picker="">
```

I tried out included minified version, but still no success, so it seems that this is not the problem. Any ideas what could went wrong?

Thanks in advance, cheers!

**Comment by krtek** [comment]
Hello, 
I just tested it locally with minified version and it works OK. Are you sure there is no glitch in your code?

While minifying AngularJS code you must use [ngAnnotate](https://github.com/mzgol/grunt-ng-annotate) grunt plugin. Did you?

**Comment by szymonmichalak** [comment]
Hi,
thanks for your reply and will to help!

Yes, I am using ngAnnotate.

HTML is:
```html
<div class="date-picker ng-pristine ng-untouched ng-valid ng-isolate-scope" options="datepickerOptions" ng-model="active.date" date-range-picker="">
<!-- custom HTML for showing dates -->
</div>
```

The settings in the Controller:

```javascript
$scope.datepickerOptions = {
      ranges: {
        'Today': [moment(), moment()],
        'Yesterday': [moment().subtract(1, 'days'), moment().subtract(1, 'days')],
        'Last 7 Days': [moment().subtract(6, 'days'), moment()],
        'Last 30 Days': [moment().subtract(29, 'days'), moment()],
        'This Month': [moment().startOf('month'), moment().endOf('month')],
        'Last Month': [moment().subtract(1, 'month').startOf('month'), moment().subtract(1, 'month').endOf('month')]
      },
      format: 'L'
    };
```

I have really no idea what could possibly go wrong, but I don't preclude that there's a glitch here...

**Comment by szymonmichalak** [comment]
I've made another, simple ``yo angular-require`` app and added only ``angular-daterangepicker``.
I've played with some setups, and I got the same error whenever the dependencies were wrong (or in wrong order) in normal development. When I switched to build, problem is the same.

I checked my project and the dependencies in shim in requirejs are okay:

```
'angular-daterangepicker': [
      'jquery',
      'bootstrap',
      'angular',
      'moment',
      'bootstrap-daterangepicker'
    ]
```

So probably it must be the problem of minified requirejs file, that combines all the JS in the project...

I will try tomorrow to build my own, very simple directive for bootstrap daterangepicker and let you know. Thanks!
